enterprise/replay/dumpsloader.go

```diff
@@ -142,40 +143,35 @@ func (handle *GWReplayRequestHandler) fetchDumpsList(handleDump *DumpsLoaderHand
 		}
 
 		pkgLogger.Info("Total gw_dumps count from S3 : ", len(objects))
-		err = handleDump.DbHandle.Store(context.TODO(), jobs)
+		err = gwHandle.handle.dbHandle.Store(ctx, jobs)
 		if err != nil {
-			panic(fmt.Errorf("Failed to write gw_dumps locations to DB with error: %w", err))
-		}
-
-		if !s3Objects.Truncated {
-			break
+			panic(fmt.Errorf("failed to write gw_dumps locations to DB with error: %w", err))
 		}
 	}
 
 	pkgLogger.Info("Dumps loader job is done")
-	handleDump.done = true
+	gwHandle.handle.done = true
 }
 
-func (handle *ProcErrorRequestHandler) fetchDumpsList(handleDump *DumpsLoaderHandleT) {
-	var continuationToken *string
+func (procHandle *ProcErrorRequestHandler) fetchDumpsList(ctx context.Context) {
 	for {
 		objects := make([]dbObjectT, 0)
 		pkgLogger.Info("fetching files list")
-		s3Objects, err := fileuploader.ListFilesWithPrefix(handleDump.prefix, handleDump.bucket, handleDump.startAfterKey, continuationToken)
+		s3Objects, err := procHandle.handle.uploader.ListFilesWithPrefix(ctx, procHandle.handle.prefix, 1000)
```

Comment:
it is not evident how we are now making sure that during the next iteration of this `for` loop, the next `1000 `items are going to be fetched